BugFix: fixup failing MacOs CI builds

SlySven opened the pull request:
~~Inserts some symbolic links to account for the change of the library name for linking from `zziplib` to `zziplib-0` on MacOs!~~

Edit: rendered unnecessary as I have now excised the use of the Lua **[`luazip`](https://github.com/mpeterv/luazip)** module (in favour of **[`lua-zip`](https://github.com/brimworks/lua-zip)**) on MacOs to provide the `zip` module in the Lua sub-system and THAT does not have a dependency on `zziplib` (a.k.a. `zzip` a.k.a. `libzzip`)

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: fixup failing MacOs CI builds

Inserts some symbolic links to account for the change of the library name
for linking from `zziplib` to `zziplib-0` on MacOs!

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: add a missing double-quote from previous commit

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: put some debug output lines back in

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: fix a typo in previous commit

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: restore manual qmake LIBS addition [skip appveyor]

It shouldn't be needed but it did work in the other PR where I was hacking
around trying to solve the MacOs CI build failures and it is the only
functional difference I could see.

It is not expected that the second (the CMake one) build job will pass.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: correct error in numeric value use in file names

Also reremove the manual insertion of the libraries into the QMake LIBS
variable - as it should be done by Qt and pkg-config automagically.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: add zziblib to main project MacOs CMake libraries list

Hopefully this will fix the second (CMake) Travis CI build job.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: try a different name for the zziblib libary

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: try adding a FindZZip.cmake file to aid finding the zziplib package

The added file is Public Domain licensed from the "Ogre" (Object-oriented
Graphics Rendering Engine) project at http://www.ogre3d.org/. The file was
sourced from:
https://github.com/OGRECave/ogre/blob/master/CMake/Packages/FindZZip.cmake

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Append: add a missing file of macros for use with previously added file

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert last four commits

SlySven committed:
Refactor: remove embedded luazip code and switch to lua-zip rock on MacOs

Reviewing the AppVeyor builds for Windows it appears that we are repeating
this use of luazip instead of lua-zip there as well. As there is some
tweaking and the compilation is done manually I have left comments about
this in the relevant CI files...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: remove a couple of remnants of luazip left behind from last commit

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert: don't change the Windows building of lua-zip right now...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
CodeFactor likes you a lot now that you've deleted luazip.h!

SlySven commented:
Hah!

Were *you* aware that we have been using the `luazip` luarock and not the `lua-zip` one? It is the former that uses `zziplib` but the latter does not... hence the most recent CI build (where I removed the source of the former from being included in `TLuaInterpreter.cpp`) has finally compiled. OTOH I've broken the Windows build by fiddling with a luazip to lua-zip switch - so I'd better put it back there...

:juggling_person:

SlySven committed:
Revert: another bit of don't change the Windows building of lua-zip now

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert: put back in a needed library for MacOs QMake builds

The link flag `-lz` is for the zlib library {https://zlib.net/}...

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented:
Not off the top of my head. The less complexity, the better - less can break. I totally welcome the removal of things we don't need :tada:

vadi2 commented:
Linux works well, but not macOS:

![image](https://user-images.githubusercontent.com/110988/79640000-6bf8df00-818f-11ea-9bb7-c5d14b7714b2.png)

SlySven commented:
That was what I *sort* of expected - can you install the `lua-zip` rock (note the hyphen) and copy the `zip.so` file to the same directory as the mudlet executable (the `./Mudlet.app/Contents/MacOS/` directory in the app bundle) and see if it then works...

SlySven commented:
Also, that *should* be addressed by: https://github.com/Mudlet/installers/pull/72 ...

vadi2 commented:
Cool! Merged that one and kicked this one to rebuild.

SlySven commented:
So plonking that `zip.so` in sorted it out? :pray:

vadi2 commented:
I haven't tried that, I'll try this build when it's complete.

SlySven commented:
Ah, no, the MacOS Qmake build ended up with:
```sh
INFO | macdeployqtfix terminated with success

cp: /Users/travis/.luarocks/lib/lua/5.1/brimworks/yajl.so: No such file or directory
Done. Your build exited with 0.
```
Can you check to see where it goes if you do a `luarock --local install lua-zip` ...

*EDit: oh, `yajl.so` from `lua-yajl` not `zip.so` from `lua-zip`,* \*sigh\*!

vadi2 commented:
Hm... nowhere. It looks like the development environment on the machine broke:

![Selection_535](https://user-images.githubusercontent.com/110988/79640906-a0bb6500-8194-11ea-954b-7998a7bcd954.png)

Could you do it in Travis?

SlySven commented:
:mag_right:  You are building your luarock against Lua 5.**3** not 5.**1**!
 
Well the build log on that last run for the MacOS QMake (Job 1) looks okay:
```sh
Installing https://luarocks.org/lua-yajl-2.0-1.src.rock
lua-yajl 2.0-1 depends on lua >= 5.1 (5.1-1 provided by VM)
lua_yajl.c:146:1: warning: control reaches end of non-void function [-Wreturn-type]
}
^
1 warning generated.

env MACOSX_DEPLOYMENT_TARGET=10.8 gcc -O2 -fPIC -I/usr/local/opt/lua@5.1/include/lua5.1 -c lua_yajl.c -o lua_yajl.o
env MACOSX_DEPLOYMENT_TARGET=10.8 gcc -bundle -undefined dynamic_lookup -all_load -o yajl.so lua_yajl.o -lyajl
lua-yajl 2.0-1 is now installed in /Users/travis/.luarocks (license: MIT/X11)
Installing https://luarocks.org/lua-zip-0.2-0.src.rock
lua-zip 0.2-0 depends on lua (5.1-1 provided by VM)
env MACOSX_DEPLOYMENT_TARGET=10.8 gcc -O2 -fPIC -I/usr/local/opt/lua@5.1/include/lua5.1 -c lua_zip.c -o lua_zip.o -I/usr/local/include
env MACOSX_DEPLOYMENT_TARGET=10.8 gcc -bundle -undefined dynamic_lookup -all_load -o brimworks/zip.so lua_zip.o -L/usr/local/lib -Wl,-rpath,/usr/local/lib -lzip
lua-zip 0.2-0 is now installed in /Users/travis/.luarocks (license: MIT)

```

On the other hand, examining the build `.dmg` (a bit tricky on a Linux machine) it seems that the `Mudlet` executable shares it's directory with:
* lcf (folder/sub-directory)
* luasql (folder/sub-directory)
* lfs.so
* lua-utf8.so
* rex_pcre.so
* yajl.so

But there is no sign of a `zip.so` which is thus AWOL...

SlySven commented:
Ah, I think we might be installing lua-yajl twice (one in the main build, as it is needed for compilation - processing the translation statistics; and one in the installer as it is needed for the bundle).

Yes! I see it - I was copying the lua `zip.so` from the wrong location - I had added a redundant copy of the `yajl.so` as well but as that was being done from the correct place further down the `./osx/make-install.sh` file in the installer it was irrelevant - just rolling a fix now, it is: https://github.com/Mudlet/installers/pull/73 !!!

SlySven commented:
Just kicked the number 1 Travis CI build job to rebuild a new .dmg...

SlySven committed:
Testing: find macOs qmake CI build placement of zip luarock [skip appveyor]

Temporarily use a Testing PR in the mudlet/installers repo

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: correct location of temporary installer branch [skip appveyor]

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
BugFix: temp installer to get lua-zip module right on MacOs [skip appveyor]

Will need a further commit to revert to standard installer repo when this
is shown to put the `zip.so` module in the right place for MacOs CI /
installer builds and the installer repo is updated to include the change.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revert: restore C loading of `zip.so` for MacOs case. [skip appveyor]

The reason for this code is not clear - why is it needed on MacOs and not
other ones...?

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: load zip.so from a sub-directory on MacOS [skip appveyor]

Because the module is stored as brimworks/zip.so (to differentiate it from
the other zip.so that we are replacing for MacOs builds) we have to bundle
it in such a sub-directory. This also needs a tweak to the installer but
that is already in place for a testing run.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: take out a code fragment again [skip appveyor]

I shouldn't have put it in as it wasn't relevant and now break things.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 committed:
Revert back to main repo of installers

vadi2 reviewed (approved):
Thanks a lot for patching this up! This was really tough to do. Cheers :beer: 

![Selection_537](https://user-images.githubusercontent.com/110988/79679584-46a7b780-8207-11ea-9b45-dbf66aafb17f.png)

Also another one that deletes more than it adds. Awesome :wink:

vadi2 commented:
Merging it in for you in order to update all PRs affected by this.

SlySven commented:
I really hope you did a squash rather than a rebase merge - there is an awful lot of c\*\*p in that chain of commits and they really ought not to be shown in the end result - oh you did - but all that mess is present in the commit message whereas I would have summarised it to to prune out the unnecessary stuff about `FindZzipLib.cmake` which is now total pointless....! :facepalm:

Okay - FTR this means that MacOS developers compiling from sources will need to have the `lua-zip` rock in their repositories rather than the `luazip` one.

I need to re-tweak that `TLuaInterpreter::loadModule(...)` addition + refactoring I did recently so as to try and load BOTH of them for ALL OSs and only moan if *neither* are found...

... I'll try and do that in the next couple of days.

vadi2 commented:
Why both? Only dealing with one will keep our code simpler (and thus less bugs)

SlySven commented:
Yeah, but it means that instead of all OSes using the same module they use different ones and that will have to be documented somewhere - it is easier for the end user/hacker if they can source the one that works easier for them and makes things hard for us to debug a problem if the two modules do not behave **exactly** the same under ALL circumstances and we cannot swap between them just by changing the luarocks we have installed...

vadi2 commented:
Hmm so you're saying yes we should have one because one will have the same behaviour? It's a bit unclear

SlySven commented:
It took me a while to work out that the name you give for a luarock need not be the name of the library/module that it installs and there can be compatible rocks that do the same thing. E.g.:
* `luazip` {`./zip.so`} vs. `lua-zip` {`./brimworks/zip.so}`
* `luautf8` {`./lua-utf8`} vs. `utf8` {`./utf8.so`}

Anyhow that is now ready and works for me (on Linux) see: https://github.com/Mudlet/Mudlet/pull/3648 .
